src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -0,0 +1,1899 @@
+
+            // Set up our return value:
+            alignedAlleleSeq = substitutedAlignedAlleleSeq;
+        }
+
+        return alignedAlleleSeq;
+    }
+
+    /**
+     * Get the aligned coding sequence for the given reference allele.
+     * @param referenceSnippet {@link String} containing a short excerpt of the reference sequence around the given reference allele.
+     * @param referencePadding Number of bases in {@code referenceSnippet} before the reference allele starts.  This padding exists at the end as well (plus some other bases to account for the length of the alternate allele if it is longer than the reference).
+     * @param refAllele The reference {@link Allele}.
+     * @param codingSequenceRefAlleleStart The position (1-based, inclusive) in the coding sequence where the {@code refAllele} starts.
+     * @param alignedRefAlleleStart The in-frame position (1-based, inclusive) of the first base of the codon containing the reference allele.
+     * @return A {@link String} of in-frame codons that contain the entire reference allele.
+     */
+    public static String getAlignedRefAllele(final String referenceSnippet,
+                                             final int referencePadding,
+                                             final Allele refAllele,
+                                             final int codingSequenceRefAlleleStart,
+                                             final int alignedRefAlleleStart ) {
+
+        Utils.nonNull(referenceSnippet);
+        Utils.nonNull(refAllele);
+
+        final int extraBasesNeeded = (codingSequenceRefAlleleStart - alignedRefAlleleStart);
+        int refStartPos = referencePadding - extraBasesNeeded;
+
+        if ( refStartPos < 0 ) {
+            refStartPos = 0;
+        }
+
+        // Round to the nearest multiple of 3 to get the end position.
+        int refEndPos = refStartPos + (int)(Math.ceil((extraBasesNeeded + refAllele.length()) / 3.0) * 3);
+
+        // Create the aligned reference:
+        String alignedReferenceAllele = referenceSnippet.substring(refStartPos, refEndPos);
+
+        // Make sure our reference is what we expect it to be with the ref allele:
+        if ( !alignedReferenceAllele.substring(extraBasesNeeded, extraBasesNeeded + refAllele.length()).equals(refAllele.getBaseString()) ) {
+            // Oh noes!
+            // Ref allele is different from reference sequence!
+
+            // Oh well, we should use the reference we were given anyways...
+            final String substitutedReferenceSnippet = getAlternateSequence(referenceSnippet, referencePadding + 1, refAllele, refAllele);
+            refEndPos = refStartPos + (int)(Math.ceil((extraBasesNeeded + refAllele.length()) / 3.0) * 3);
+
+            final String substitutedAlignedAlleleSeq = substitutedReferenceSnippet.substring(refStartPos, refEndPos);
+
+            // Warn the user!
+            logger.warn("
```

Comment:
This might be a good place for a builder, since I think you will have places where the SequenceComparison is in a bad state.  And maybe we can get rid of all the field null checks?

Reply:
@LeeTL1220 I'm not sure what this is in reference to - I think the comments are somehow out of place.